src/framework/global/serialization/xmldom.h

```diff
@@ -66,12 +68,35 @@ class XmlDomElement : public XmlDomNode
 
     String text() const;
 
+    XmlDomAttribute firstAttribute() const;
+    XmlDomAttribute attribute(const char* name) const;
+
 private:
 
     friend class XmlDomDocument;
     friend class XmlDomNode;
 
-    XmlDomElement(const std::shared_ptr<XmlDomData>& data, uintptr_t node);
+    XmlDomElement(const std::shared_ptr<XmlDomImplData>& data, uintptr_t node);
+};
+
+class XmlDomAttribute
+{
+public:
+
+    bool isNull() const;
+    String attributeName() const;
+
+    String value() const;
```

Comment:
Is it an idea to add `toInt`/`toFloat`/`toDouble` here, for convenience?

Reply:
I was not planning to, since the `String` class already has them all. It felt redundant because one has to ask oneself where is the cutoff point. How much of the `String` api deserves convenience pass-thru functions? But it could be done.